app/src/main/java/com/github/multimatum_team/multimatum/viewmodel/DeadlineListViewModel.kt

```diff
@@ -58,32 +59,38 @@ class DeadlineListViewModel @Inject constructor(
     /**
      * Re-fetch all deadlines from the repository and assign the value to the LiveData.
      */
-    private fun refreshDeadlines(callback: (Map<DeadlineID, Deadline>) -> Unit) = viewModelScope.launch {
-        _deadlines.value = deadlineRepository.fetchAll()
-        callback(_deadlines.value!!)
-    }
+    private fun refreshDeadlines(callback: (Map<DeadlineID, Deadline>) -> Unit = {}) =
+        viewModelScope.launch {
+            _deadlines.value = deadlineRepository.fetchAll()
+            callback(_deadlines.value!!)
+        }
 
     /**
      * Get all deadlines.
      */
     fun getDeadlines(): LiveData<Map<DeadlineID, Deadline>> =
         _deadlines
 
+    fun getDeadline(id: DeadlineID): Deadline =
+        _deadlines.value!![id]!!
```

Comment:
The first `!!` can fail if `_deadlines` is not yet initialized. The only way this can happen is that `getDeadline` is called immediately after the app started which is possible but unlikely. The second `!!` should never fail if the `id` exists, which should always be the case if we use the IDs provided by the repository.